clippy_lints/src/redundant_field_names.rs

```diff
@@ -36,10 +38,14 @@ impl LintPass for RedundantFieldNames {
 
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for RedundantFieldNames {
     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr) {
-        if let ExprStruct(_, ref fields, _) = expr.node {
+        if let ExprStruct(ref path, ref fields, _) = expr.node {
```

Comment:
Have you tried checking with the `in_macro` method? I think this should have expansion info, so we should be able to detect it that way

Reply:
I didn't know `in_macro()` until now.
But the method intends not to treat range expressions as "in_macro" (since #1373).

https://github.com/rust-lang-nursery/rust-clippy/blob/598acba7d5561b2096dfa72a39247983852b9bea/clippy_lints/src/utils/mod.rs#L56-L65

So I've added a method which determines whether an expression is a range expression.
Now range expressions are ignored, and hand-written `Range` family structs are treated normally by the lint.

Reply:
nice! The diff lgtm. Let's wait for travis and then merge